sys/include/usb/usbus.h

```diff
@@ -0,0 +1,533 @@
+    usbus_string_t product;                         /**< Product string                        */
+    usbus_string_t config;                          /**< Configuration string                  */
+    usbus_endpoint_t ep_out[USBDEV_NUM_ENDPOINTS];  /**< USBUS OUT endpoints                   */
+    usbus_endpoint_t ep_in[USBDEV_NUM_ENDPOINTS];   /**< USBUS IN endpoints                    */
+    event_queue_t queue;                            /**< Event queue                           */
+    usbdev_t *dev;                                  /**< usb phy device of the usb manager     */
+    usbus_handler_t *control;                       /**< Ptr to the control endpoint handler   */
+    usbus_hdr_gen_t *hdr_gen;                       /**< Top level header generators           */
+    usbus_string_t *strings;                        /**< List of descriptor strings            */
+    usbus_interface_t *iface;                       /**< List of USB interfaces                */
+    usbus_handler_t *handlers;                      /**< List of event callback handlers       */
+    uint32_t ep_events;                             /**< bitflags with endpoint event state    */
+    kernel_pid_t pid;                               /**< PID of the usb manager's thread       */
+    uint16_t str_idx;                               /**< Number of strings registered          */
+    usbus_state_t state;                            /**< Current state                         */
+    usbus_state_t pstate;                           /**< state to recover to from suspend      */
+    uint8_t addr;                                   /**< Address of the USB peripheral         */
+};
+
+/**
+ * @brief Submit an event to the usbus thread
+ *
+ * @param   usbus   USBUS context
+ * @param   event   event to post
+ */
+static inline void usbus_event_post(usbus_t *usbus, event_t *event)
+{
+    event_post(&usbus->queue, event);
+}
+
+/**
+ * @brief Add a string descriptor to the USBUS thread context
+ *
+ * @param[in] usbus   USBUS context
+ * @param[in] desc    string descriptor context
+ * @param[in] str     C string to use
+ *
+ * @return            Index of the string descriptor
+ */
+uint16_t usbus_add_string_descriptor(usbus_t *usbus, usbus_string_t *desc,
+                                     const char *str);
+
+/**
+ * @brief Add an interface to the USBUS thread context
+ *
+ * @param[in] usbus   USBUS context
+ * @param[in] iface   USB interface to add
+ *
+ * @return          interface index
+ */
+uint16_t usbus_add_interface(usbus_t *usbus, usbus_interface_t *iface);
+
+/**
+ * @brief Add an endpoint to the specified interface
+ *
+ * An @ref usbdev_ep_t is requested from the low level peripheral matching the
+ * type, direction and buffer length.
+ *
+ * @param[in] usbus USBUS context
+ * @param[in] iface USB interface to add the endpoint to
+ * @param[in] ep    USB endpoint struct to use
```

Comment:
This param is not used by the function, should be removed

Reply:
removed